shared/src/main/kotlin/org/jetbrains/kotlin/konan/util/Util.kt

```diff
@@ -57,3 +60,69 @@ fun String.removeSuffixIfPresent(suffix: String) =
     if (this.endsWith(suffix)) this.dropLast(suffix.length) else this
 
 fun <T> Lazy<T>.getValueOrNull(): T? = if (isInitialized()) value else null
+
+
+
+/// we use the fake delimiter to support the feature
+/// of the [@Argument] parser to merge several arguments
+/// into the same argument (it works only for [Array<String>]
+/// type, but we do not need the separator
+///
+/// the right fix is to update the [@Argument] to support paths separation
+const val ARGUMENT_NO_DELIMITER = "\\n\\t\\t\\n\\t\\t\\n\\ue000\\ue001\\ue002\\n\\t\\t\\t\\t\\n"
+
+fun parseCommandLineString(value: () -> Array<String>) = object: ReadOnlyProperty<Any, Array<String>> {
+    override fun getValue(thisRef: Any, property: KProperty<*>): Array<String> {
+        return value().flatMap { parseCommandLineString(it) }.toTypedArray()
+    }
+}
+
+fun escapeToCommandLineString(args: Iterable<String>) : String {
+    return args.joinToString(separator = " ", transform = {
+        when {
+            it.startsWith("\'") && it.endsWith("\'") -> it
+            it.startsWith("\"") && it.endsWith("\"") -> it
+            it.contains(" ") -> "\"$it\""
+            it.isEmpty() -> "\"\""
+            //TODO: possible incorrect case for "a b c\" (slash escapes ")
+            else -> it
+        }
+    })
+}
+
+fun parseCommandLineString(cmdString: String): List<String> {
+    //inspired by IntelliJ IDEA source code (Apache 2.0 by JetBrains)
+    //com.intellij.openapi.util.text.StringUtilRt#splitHonorQuotes
+
+    val result = mutableListOf<String>()
+    val builder = StringBuilder(cmdString.length)
+
+    var inQuotes = false
+    for (i in 0 until cmdString.length) {
+        val c = cmdString[i]
+        if (c == ' ' && !inQuotes) {
+            if (builder.isNotEmpty()) {
+                result.add(builder.toString())
+                builder.setLength(0)
+            }
+            continue
+        }
+        if ((c == '"' || c == '\'') && !(i > 0 && cmdString[i - 1] == '\\')) {
```

Comment:
This parser fails to handle a lot of complicated cases, e.g. `' " abc "  '`.
I consider implementing escaping support harmful unless it has reliable widely recognized behaviour.

Reply:
In general, the way how arguments are parsed by a process depends on the process. It depends on the OS. You may have the same commandline processed differently by java, cmd, or bash.
I think the general idea to pass quoted commandline string is broken from the very beginning. 

Alternative could be to add yet anothe sub-command to pass exactly one argument, e.g. 
```-linkerOpt "abc" -linkerOpt `' " abc " '````
Both commands may blindly include the whole value as one array element. Simliarly we may do for the `.def` files, but it may require to patch the way we use `Properties` class in the implementation


What do you think?

Reply:
> Alternative could be to add yet anothe sub-command to pass exactly one argument,

How is it different from the solution I proposed in https://github.com/JetBrains/kotlin-native/pull/2684#discussion_r257906960 ?